src/runtime.js

```diff
@@ -7,10 +7,11 @@ var Obj = require('./object');
 // we know how to access variables. Block tags can introduce special
 // variables, for example.
 var Frame = Obj.extend({
-    init: function(parent) {
+    init: function(parent, inMacro) {
```

Comment:
What do you think about using a more generic name for this new parameter; something like `isolate`? It seems likely to me that this same parameter might prove useful in solving the other portions of #561, in which case the name `inMacro` might no longer apply.